tools/wptrunner/wptrunner/stability.py

```diff
@@ -91,9 +106,18 @@ def is_inconsistent(results_dict, iterations):
     return len(results_dict) > 1 or sum(results_dict.values()) != iterations
 
 
+def is_slow(test):
+    # We are interested in tests that almost time out (i.e. likely to be
+    # flaky). Tests that consistently time out are not considered "slow".
+    if test["status"].keys() == ['TIMEOUT']:
```

Comment:
I think `CRASH` should also be excluded because we can change the status from `TIMEOUT` to `CRASH` if we detect evidence of a crash after the test finished. In general I wonder if testing 

Also, it seems arguable that the handling of a test that sometimes times out and sometimes doesn't is wrong; if we get `PASS`, `PASS`, `PASS`, `TIMEOUT`, `PASS` then we'll call the test slow. But if the passes take 1s and the timeout is 10s then we are probably seeing a race condition that causes the test to hang sometimes, and the obvious solution to a slow test (Add a longer timeout) isn't going to help. So I think we should either store the duration per result and filter here, or only update the duration when we see a non-TIMEOUT, non-CRASH result. The latter is easier, but the former could be used to provide more detailed logging if we want.

Reply:
You're right about `CRASH`.

As for the second example (inconsistent results), it won't actually happen as we call `is_inconsistent` before `is_slow`, so such tests will be considered as flaky instead of slow. But I do see the code can be confusing and error-prone as the prerequisite isn't obvious. I'll tweak it a bit.